src/Microsoft.AspNet.Http/Features/QueryFeature.cs

```diff
@@ -41,11 +41,23 @@ void IFeatureCache.CheckFeaturesRevision()
         {
             if (_cachedFeaturesRevision != _features.Revision)
             {
-                _request = null;
-                _cachedFeaturesRevision = _features.Revision;
+                ResetFeatures();
             }
         }
 
+        public void UpdateFeatures(IFeatureCollection features)
```

Comment:
This doesn't make sense, UpdateFeatures is never called on any of these Feature objects, they just get nulled out.